index.js

```diff
@@ -108,6 +108,13 @@ var xml = function(options) {
           companion.element(r.type, (r.creativeType) ? { creativeType : r.creativeType } : {}).cdata(r.uri);
           if (r.adParameters) companion.element('AdParameters', r.adParameters.data, { xmlEncoded : r.adParameters.xmlEncoded });
         });
+        if (c.trackingEvents.length > 0) var trackingEvents = companion.element('TrackingEvents');
```

Comment:
Est-ce que **trackingEvents** est toujours défini dans **c** ? car s'il ne l'est pas le test suivant va crasher :
```c.trackingEvents.length > 0```

Reply:
Bien vu ! Je rajoute la condition :
```suggestion
        if (c.trackingEvents && c.trackingEvents.length > 0) var trackingEvents = companion.element('TrackingEvents');
```